src/cargo/util/toml.rs

```diff
@@ -995,13 +1088,20 @@ struct TomlTarget {
     required_features: Option<Vec<String>>,
 }
 
-#[derive(Deserialize, Clone)]
-#[serde(untagged)]
+#[derive(Clone)]
 enum PathValue {
     String(String),
     Path(PathBuf),
 }
 
+impl de::Deserialize for PathValue {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where D: de::Deserializer
+    {
+        Ok(PathValue::String(String::deserialize(deserializer)?))
+    }
+}
```

Comment:
If we are writing a custom deserialization anyway, maybe we can remove `String` variant from the enum?